src/rospack_backcompat.cpp

```diff
@@ -40,7 +40,8 @@ namespace rospack
 int
 ROSPack::run(int argc, char** argv)
 {
-  rospack::Rospack rp;
+  // allow caching of results between calls (of same process)
+  static rospack::Rospack rp;
```

Comment:
This change has the chance to change the current behavior. How can you be sure that the caller want's to reuse the previous instance? Eventually the caller relies on the "new" crawl to have a different result.

Therefore similar changes have added the option to pass in a rospack instance instead, e.g. ros/ros_comm#682.